src/Microsoft.ML.StandardLearners/Standard/Online/OnlineLinear.cs

```diff
@@ -20,24 +20,40 @@ namespace Microsoft.ML.Trainers.Online
 
     public abstract class OnlineLinearArguments : LearnerInputBaseWithLabel
     {
+        /// <summary>
+        /// Number of training iterations through the data.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Number of iterations", ShortName = "iter", SortOrder = 50)]
         [TGUI(Label = "Number of Iterations", Description = "Number of training iterations through data", SuggestedSweeps = "1,10,100")]
         [TlcModule.SweepableLongParamAttribute("NumIterations", 1, 100, stepSize: 10, isLogScale: true)]
         public int NumIterations = OnlineDefaultArgs.NumIterations;
 
+        /// <summary>
+        /// Initial weights and bias, comma-separated.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Initial Weights and bias, comma-separated", ShortName = "initweights")]
         [TGUI(NoSweep = true)]
         public string InitialWeights;
```

Comment:
make it internal, mark in Argument as CommandLineOnly, and create another parameter (or actually two parameters) which would accept array of floats (and single float for bias).

Reply:
you're right, but this is a non-trivial change. our current API takes this as a string. could you please file your proposal as an issue in Project 13.

---
In reply to: [255264280](https://github.com/dotnet/machinelearning/pull/2483#discussion_r255264280) [](ancestors = 255264280)